Sources/FlowCommoniOSFiles/Animation.swift

```diff
@@ -77,49 +84,65 @@ open class Animation: NSObject, CAAnimationDelegate {
         layer.timeOffset = 0.0
         layer.beginTime = 0.0
         let timeSincePause = layer.convertTime(CACurrentMediaTime(), from: nil) - pausedTime
+
+        for keyframeAnimation in keyframeAnimations {
+            keyframeAnimation.timeOffset = 0
+        }
         layer.beginTime = timeSincePause
     }
 
     /// Pauses the animation.
     open func pause() {
-        let pausedTime = layer.convertTime(CACurrentMediaTime(), from: nil)
-        offset(to: pausedTime)
+        offset(to: layer.convertTime(CACurrentMediaTime(), from: nil))
+        layer.speed = 0
     }
 
     /// Resets the animation to time 0.
-    open func reset() {
+    open func reset(notify: Bool = true) {
         CATransaction.suppressAnimations {
             layer.removeAllAnimations()
-            layer.beginTime = 0
-            offset(to: 0)
-
+            
             for keyframeAnimation in keyframeAnimations {
                 layer.setValue(keyframeAnimation.values?.first, forKeyPath: keyframeAnimation.keyPath!)
             }
 
-            addAllAnimations()
+            offset(to: 0)
+            self.addAllAnimations(notify: notify)
             layer.speed = 0
         }
     }
 
     /// Adds all the animations to `layer` so they can be played.
-    private func addAllAnimations() {
+    private func addAllAnimations(notify: Bool = true) {
         DispatchQueue.main.async { [weak self] in
             guard let keyframeAnimations = self?.keyframeAnimations, let layer = self?.layer else {
                 return
             }
             for keyframeAnimation in keyframeAnimations {
                 layer.add(keyframeAnimation, forKey: keyframeAnimation.keyPath)
             }
+            //Need to have a notification because adding animations happens asynchronously
+            if notify {
+                NotificationCenter.default.post(name: Animation.Defaults.addAllAnimationsNotification, object: self)
```

Comment:
I don't think you need a notification mechanism, I think you can just tell the delegate directly and just call 
` delegate?.ready(animation: self)` directly here.

Reply:
onCompletion?()